Headlines/src/Categories/UIImageEffects.swift

```diff
@@ -0,0 +1,281 @@
+    
+    public func applyTintEffectWithColor(_ tintColor: UIColor) -> UIImage? {
+        let effectColorAlpha: CGFloat = 0.6
+        var effectColor = tintColor
+        
+        let componentCount = tintColor.cgColor.numberOfComponents
+        
+        if componentCount == 2 {
+            var b: CGFloat = 0
+            if tintColor.getWhite(&b, alpha: nil) {
+                effectColor = UIColor(white: b, alpha: effectColorAlpha)
+            }
+        } else {
+            var red: CGFloat = 0
+            var green: CGFloat = 0
+            var blue: CGFloat = 0
+            
+            if tintColor.getRed(&red, green: &green, blue: &blue, alpha: nil) {
+                effectColor = UIColor(red: red, green: green, blue: blue, alpha: effectColorAlpha)
+            }
+        }
+        
+        return applyBlurWithRadius(10, tintColor: effectColor, saturationDeltaFactor: -1.0, maskImage: nil)
+    }
+    
+    public func applyBlurWithRadius(_ blurRadius: CGFloat, tintColor: UIColor?, saturationDeltaFactor: CGFloat, maskImage: UIImage? = nil) -> UIImage? {
+        // Check pre-conditions.
+        if (size.width < 1 || size.height < 1) {
+            print("*** error: invalid size: \(size.width) x \(size.height). Both dimensions must be >= 1: \(self)")
+            return nil
+        }
+        guard let cgImage = self.cgImage else {
+            print("*** error: image must be backed by a CGImage: \(self)")
+            return nil
+        }
+        if maskImage != nil && maskImage!.cgImage == nil {
+            print("*** error: maskImage must be backed by a CGImage: \(String(describing: maskImage))")
+            return nil
+        }
+        
+        let FLTEPSILON = CGFloat(Float.ulpOfOne)
+        let screenScale = UIScreen.main.scale
+        let imageRect = CGRect(origin: CGPoint.zero, size: size)
+        var effectImage = self
+        
+        let hasBlur = blurRadius > FLTEPSILON
+        let hasSaturationChange = fabs(saturationDeltaFactor - 1.0) > FLTEPSILON
+        
+        if hasBlur || hasSaturationChange {
+            func createEffectBuffer(_ context: CGContext) -> vImage_Buffer {
+                let data = context.data
+                let width = vImagePixelCount(context.width)
+                let height = vImagePixelCount(context.height)
+                let rowBytes = context.bytesPerRow
+                
+                return vImage_Buffer(data: data, height: height, width: width, rowBytes: rowBytes)
+            }
+            
+            UIGraphicsBeginImageContextWithOptions(size, false, screenScale)
+            guard let effectInContext = UIGraphicsGetCurrentContext() else { return  nil }
+            
+            effectInContext.scaleBy(x: 1.0, y: -1.0)
+            effectInContext.translateBy(x: 0, y: -size.height)
+            effectInContext.draw(cgImage, in: imageRect)
+            
+            var effectInBuffer = createEffectBuffer(effectInContext)
+            
+            
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/b49c86a9b05bec3bbe17833b5a716a8293c42cf1/Headlines/src/Categories/UIImageEffects.swift#L172"></span>Limit vertical whitespace to a single empty line. Currently 2.</td>
    </tr>
  </tbody>
</table>